shardingsphere-infra/shardingsphere-infra-optimize/src/test/resources/sales/t_order_item_federate_sharding.csv

```diff
@@ -0,0 +1,21 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
```

Comment:
Why do we need a csv database? Do you think `h2` will be simpler?

Reply:
@tristaZero do you mean use jdbc adaptor instead of csv adaptor? As csv adaptor and jbdc adaptor both in the calcite. I simplified the csv adaptor to make it easier to be used in the test. Because we should create Schema in the init() method.
```
    public void init() {
        URL path = this.getClass().getClassLoader().getResource("");
        if (null != path) {
            String url = path.getPath() + "/sales";
            Schema calciteSchema = new LocalSchema(new File(url));
            Map<String, ShardingSphereMetaData> metaDataMap = createMetaDataMap();
            OptimizeContextFactory optimizeContextFactory = new OptimizeContextFactory(metaDataMap);
            OptimizeContext context = optimizeContextFactory.create("testSchema", calciteSchema);
            optimizer = new ShardingSphereOptimizer(context);
        }
    }
```
If change from csv to jdbc, it is complicated to initialize a Schema by jdbc. Bellow is csv and jdbc constructor:
CSV
```
    /**
     * Creates a CSV schema.
     * @param directoryFile Directory that holds csv files
     */
    public LocalSchema(final File directoryFile) {
        super();
        this.directoryFile = directoryFile;
        tableMap = createTableMap();
    }
```
JDBC
```
    public JdbcSchema(DataSource dataSource, SqlDialect dialect, JdbcConvention convention, String catalog, String schema) {
        this(dataSource, dialect, convention, catalog, schema, (ImmutableMap)null);
    }

    private JdbcSchema(DataSource dataSource, SqlDialect dialect, JdbcConvention convention, String catalog, String schema, ImmutableMap<String, JdbcTable> tableMap) {
        this.dataSource = (DataSource)Objects.requireNonNull(dataSource);
        this.dialect = (SqlDialect)Objects.requireNonNull(dialect);
        this.convention = convention;
        this.catalog = catalog;
        this.schema = schema;
        this.tableMap = tableMap;
        this.snapshot = tableMap != null;
    }
```